### PR TITLE
Update efo-mondo-mapping.tsv file with NCIT codes

### PR DESCRIPTION
**Pull request by sangeetashukla**
### Purpose/implementation Section
1) Add NCIT codes for the current list of cancer groups
2) Review existing EFO and MONDO codes in the file

#### What was your approach?
Review EFO, MONDO, and NCIT codes for cancer group ontologies found in the v10 histologies.

#### What GitHub issue does your pull request address?
[Issue 261](https://github.com/PediatricOpenTargets/ticket-tracker/issues/261)

#### Which areas should receive a particularly close look?

NCIT code for 'CNS Embryonal tumor' needs to be reviewed/updated since the exact ontology is not found on [OLS](https://www.ebi.ac.uk/ols/index)

#### Is there anything that you want to discuss further?
Note:

- One or both of EFO and MONDO codes were updated for below cancer groups, as found on OLS.
Atypical choroid plexus papilloma
Choroid plexus cyst
Diffuse midline glioma
Extraventricular Neurocytoma
Germinoma;Teratoma
High-grade glioma/astrocytoma
Langerhans Cell histiocytosis
Malignant peripheral nerve sheath tumor
Neurocytoma
Rhabdomyosarcoma
Rosai-Dorfman disease
Wilms tumor

- 'Metastatic   secondary tumors;Neuroblastoma' is now removed from the file since a separate 'Neuroblastoma' entry already exists.
- Codes used for 'Neurocytoma' are specifically for 'cerebral ventricle cancer' since exact ontology match was not found.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
TSV (.tsv) file is included under `results` directory.

#### What is your summary of the results?
With every update to the file, I recommend we review all the existing codes, since OLS seems to also update codes and ontologies frequently.

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by afarrel** [comment]
Hi @sangeetashukla Please consider the below suggestions. Double check the suggestions/concerns and also check synonyms of the terms:

Adenoma Mondo- 0004972 (why is it 0002454)
Atypical choroid plexus papilloma- MONDO_0002684
Central Neurocytoma- monda: MONDO_0019134  (MONDO_0002714 is for central nervous system cancer)
Choroid plexus cyst: mondo- MONDO_0022735
Should CNS Embryonal tumor- Mondo code be: MONDO_0018843? Also for EFO code
Should CNS Neuroblastoma- Mondo code be: MONDO_0005072? Ro be consistent with EFO and NCIT code.
Craniopharyngioma- Mondo: MONDO_0018907
Desmoplastic infantile astrocytoma and ganglioglioma- mondo: MONDO_0016731
Diffuse midline glioma- NCIT should be: NCIT_C129309, consider EFO: EFO_1000026
Extraventricular Neurocytoma mondo- MONDO_0016727
Germinoma- Mondo: MONDO_0002598 NCIT: NCIT_C8712 EFO: consider MONDO_0003000
Teratoma- EFO: Consider MONDO_0002601
High-grade glioma/astrocytoma- EFO: Consider MONDO_0016680
Langerhans Cell histiocytosis- MONDO: MONDO_0018310
Malignant peripheral nerve sheath tumor- Mondo: MONDO_0017827
Metastatic secondary tumors;Neuroblastoma- While redundant, include NCIT: NCIT_C3270
Neurocytoma- Consider EFO: EFO_1000856, NCIT: NCIT_C3791, MONDO: MONDO_0019134
Rosai-Dorfman disease- NCIT: NCIT_C36075, EFO: MONDO_0006412
Wilms tumor- *Couldn’t find EFO_1000056. For EFO consider: MONDO_0019004

**Comment by sangeetashukla** [comment]
> Hi @sangeetashukla Please consider the below suggestions. Double check the suggestions/concerns and also check synonyms of the terms:
> 
> Adenoma Mondo- 0004972 (why is it 0002454) Atypical choroid plexus papilloma- MONDO_0002684 Central Neurocytoma- monda: MONDO_0019134 (MONDO_0002714 is for central nervous system cancer) Choroid plexus cyst: mondo- MONDO_0022735 Should CNS Embryonal tumor- Mondo code be: MONDO_0018843? Also for EFO code Should CNS Neuroblastoma- Mondo code be: MONDO_0005072? Ro be consistent with EFO and NCIT code. Craniopharyngioma- Mondo: MONDO_0018907 Desmoplastic infantile astrocytoma and ganglioglioma- mondo: MONDO_0016731 Diffuse midline glioma- NCIT should be: NCIT_C129309, consider EFO: EFO_1000026 Extraventricular Neurocytoma mondo- MONDO_0016727 Germinoma- Mondo: MONDO_0002598 NCIT: NCIT_C8712 EFO: consider MONDO_0003000 Teratoma- EFO: Consider MONDO_0002601 High-grade glioma/astrocytoma- EFO: Consider MONDO_0016680 Langerhans Cell histiocytosis- MONDO: MONDO_0018310 Malignant peripheral nerve sheath tumor- Mondo: MONDO_0017827 Metastatic secondary tumors;Neuroblastoma- While redundant, include NCIT: NCIT_C3270 Neurocytoma- Consider EFO: EFO_1000856, NCIT: NCIT_C3791, MONDO: MONDO_0019134 Rosai-Dorfman disease- NCIT: NCIT_C36075, EFO: MONDO_0006412 Wilms tumor- *Couldn’t find EFO_1000056. For EFO consider: MONDO_0019004

Hi @afarrel 
I reviewed the codes as per your recommendations, and have made the following changes.

While other changes are good, one NCIT code (Germinoma) needs clarification.

Germinoma- Mondo: MONDO_0002598 NCIT: NCIT_C8712 EFO: consider MONDO_0003000 - Updated the EFO and MONDO codes per recommendation, but the NCIT codes for each of them are different. We would need to find the most appropriate NCIT code.


Adenoma Mondo- 0004972 (why is it 0002454) - This looks like a recent change. Last I checked the MOMDO code was not specific for thyroid adenoma. Updated now.
Atypical choroid plexus papilloma- MONDO_0002684 - Updated. 
Central Neurocytoma- monda: MONDO_0019134 (MONDO_0002714 is for central nervous system cancer)
Choroid plexus cyst: mondo- MONDO_0022735
Should CNS Embryonal tumor- Mondo code be: MONDO_0018843? Also for EFO code - 'Embryonal Neoplasm of CNS' and 'Embryonal Tumor of Central Nervous System' were some of the synonyms, and the EFO and MONDO codes for that are what I picked up. I am happy to make the change as you recommend.
Should CNS Neuroblastoma- Mondo code be: MONDO_0005072? Ro be consistent with EFO and NCIT code.
Craniopharyngioma- Mondo: MONDO_0018907
Desmoplastic infantile astrocytoma and ganglioglioma- mondo: MONDO_0016731
Diffuse midline glioma- NCIT should be: NCIT_C129309, consider EFO: EFO_1000026 - Updated. EFO seems to have been recently updated, since the MONDO code we have matches already.
Extraventricular Neurocytoma mondo- MONDO_0016727
Teratoma- EFO: Consider MONDO_0002601 - Already have that.
High-grade glioma/astrocytoma- EFO: Consider MONDO_0016680 - Updated MONDO code to match EFO code.
Langerhans Cell histiocytosis- MONDO: MONDO_0018310 - Updated
Malignant peripheral nerve sheath tumor- Mondo: MONDO_0017827 -Updated
Metastatic secondary tumors;Neuroblastoma- While redundant, include NCIT: NCIT_C3270 - Done
Neurocytoma- Consider EFO: EFO_1000856, NCIT: NCIT_C3791, MONDO: MONDO_0019134 - Updated
Rosai-Dorfman disease- NCIT: NCIT_C36075, EFO: MONDO_0006412
Wilms tumor- *Couldn’t find EFO_1000056. For EFO consider: MONDO_0019004 - Per current MONDO code notes, the old EFO code is now obsolete. New EFO code added as above.

**Comment by jharenza** [comment]
@afarrel this is ready for re-review

File: analyses/efo-mondo-mapping/results/efo-mondo-ncit-map.tsv [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+High-grade glioma/astrocytoma	MONDO_0016680	MONDO_0016680	NCIT_C102897
+Juvenile xanthogranuloma	EFO_1000311	MONDO_0015534	NCIT_C3451
+Langerhans Cell histiocytosis	EFO_1000318	MONDO_0018310	NCIT_C3107
+Low-grade glioma/astrocytoma	MONDO_0016685	MONDO_0016685	NCIT_C116342
+Malignant peripheral nerve sheath tumor	EFO_0000760	MONDO_0017827	NCIT_C3798
+Medulloblastoma	EFO_0002939	MONDO_0007959	NCIT_C3222
+Melanocytic tumor	MONDO_0003222	MONDO_0003222	NCIT_C5504
+Meningioma	Orphanet_2495	MONDO_0016642	NCIT_C3230
+Metastatic secondary tumors	EFO_0009812	MONDO_0024883	NCIT_C4968
+Metastatic secondary tumors;Neuroblastoma	EFO_0000621	MONDO_0005072	NCIT_C3270
```
Comment (jharenza) [inline analyses/efo-mondo-mapping/results/efo-mondo-ncit-map.tsv]:
```suggestion
```
We should delete this

**Review by jharenza — COMMENTED** [review]
@sangeetashukla let's keep the file the same name as before so not to break the modules :)

**Review by jharenza — COMMENTED** [review]
@sangeetashukla can you also rerun the module QC code within this PR

**Comment by sangeetashukla** [comment]
@taylordm @afarrel @jharenza 
I have updated the codes and this PR is now ready for review.

File: analyses/efo-mondo-mapping/results/efo-mondo-ncit-map.tsv (hunk 1 above)
Comment (sangeetashukla) [reply analyses/efo-mondo-mapping/results/efo-mondo-ncit-map.tsv]: @jharenza This is removed in the latest PR.

**Comment by sangeetashukla** [comment]
> @sangeetashukla let's keep the file the same name as before so not to break the modules :)

@jharenza Sure, I have removed the other file in the latest PR

**Pull request merged**